addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CharsetMismatchScanRule.java

```diff
@@ -270,7 +270,7 @@ private void raiseAlert(
                 .setOtherInfo(getExtraInfo(firstCharset, secondCharset, currentMismatch))
                 .setSolution(getSolutionMessage())
                 .setReference(getReferenceMessage())
-                .setCweId(16) // CWE-16: Configuration
+                .setCweId(351) // CWE-351: Insufficient Type Distinction
```

Comment:
[CWE-436 (Interpretation Conflict)](https://cwe.mitre.org/data/definitions/436.html) may be more appropriate here.